src/parquet/bloom_filter_algorithm.h

```diff
@@ -69,15 +73,46 @@ class BlockBasedAlgorithm : public BloomFilterAlgorithm {
       0x47b6137bU, 0x44974d91U, 0x8824ad5bU, 0xa2b7289dU,
       0x705495c7U, 0x2df1424bU, 0x9efc4947U, 0x5c6bfb31U};
 
-  bool TestBits(const uint32_t* bitset, uint32_t num_bytes, uint64_t hash) const override;
+  /// The constructor of the block-based algorithm. It allocates an 64-byte aligned
+  /// buffer as underlying bitset to facilitate SIMD instructions.
+  ///
+  /// @param num_bytes number of bytes for bitset, it should be a power of 2.
+  explicit BlockBasedAlgorithm(uint32_t num_bytes);
+
+  /// The constructor of the block-based algorithm. It copies the bitset as underlying
+  /// bitset because the given bitset may not satisfy the 64-byte alignment requirement
```

Comment:
The alignment requirement is not 64-bytes, it is either 4 bytes (as the code is now) or 32 bytes (AVX2).

Reply:
Did we discuss the alignment should be 64 bytes since the SIMD maybe trigger segfault if it access cross cache line data?